src/thread/thread_entry.cpp

```diff
@@ -553,19 +551,25 @@ thread_suspend_timeout_wakeup_and_unlock_entry (cubthread::entry *thread_p, stru
 
   if (r != 0 && r != ETIMEDOUT)
     {
+      thread_p->m_status = old_status;
+
       er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_CSS_PTHREAD_COND_TIMEDWAIT, 0);
-      return ER_CSS_PTHREAD_COND_TIMEDWAIT;
+      error = ER_CSS_PTHREAD_COND_TIMEDWAIT;
```

Comment:
```suggestion
      error = ER_CSS_PTHREAD_COND_TIMEDWAIT;
      er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
```

에러 변수를 사용하는 것이 좋아보입니다.

Reply:
반영했습니다.